pkg/apis/resource/validation/validation.go

```diff
@@ -317,7 +317,7 @@ func validateAllocationResult(allocation *resource.AllocationResult, fldPath *fi
 	var allErrs field.ErrorList
 	allErrs = append(allErrs, validateDeviceAllocationResult(allocation.Devices, fldPath.Child("devices"), requestNames)...)
 	if allocation.NodeSelector != nil {
-		allErrs = append(allErrs, corevalidation.ValidateNodeSelector(allocation.NodeSelector, fldPath.Child("nodeSelector"))...)
+		allErrs = append(allErrs, corevalidation.ValidateNodeSelector(allocation.NodeSelector, true, fldPath.Child("nodeSelector"))...)
```

Comment:
Should these be treated like the others - disallow on CREATE, allow on UPDATE only if previous was bad?

If not, can you comment on why?

Reply:
Yes, we should.